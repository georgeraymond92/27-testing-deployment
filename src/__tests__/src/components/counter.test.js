import React from "react";
import renderer from "react-test-renderer";
import Counter from "../../../components/counter/counter.js";
import { shallow, mount, render } from 'enzyme';

describe("<Counter />", () => {
  it("should return false", () => {
    let component = shallow(<Counter />);
    expect(component.find('span').exists()).toBeTruthy();
  });
  it('Should increment the state', () => {
    let component = mount(<Counter />);
    let button = component.find('.btn-green');
    button.simulate('click');
    expect(component.state('count')).toEqual(1);
  })
  it('Should decrement the state', () => {
    let component = mount(<Counter />);
    let button = component.find('.btn-red');
    button.simulate('click');
    expect(component.state('count')).toEqual(-1);
  })

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  


});
