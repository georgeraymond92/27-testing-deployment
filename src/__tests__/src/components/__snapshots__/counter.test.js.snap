// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<div
  className="deck"
>
  <div
    className="leftBtn"
  >
    <a
      className="btn btn-red down clicker"
      href="#"
      onClick={[Function]}
    >
      Decrement
    </a>
  </div>
  <div
    className="count"
  >
    <span
      className="count "
    >
      <h1>
        0
      </h1>
    </span>
  </div>
  <div
    className="btnRight"
  >
    <a
      className="btn btn-green up clicker"
      href="#"
      onClick={[Function]}
    >
      Increment
    </a>
  </div>
</div>
`;
